From 1af3a836835207dd2b0566169f314e59d93018f5 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Wed, 5 Jun 2024 10:44:56 +0200
Subject: [PATCH 1139/1141] ARM: dts: stm32: Replace OOB interrupt by IB
 interrupt for WIFI on stm32mp157c-dk2

Enabling SDIO IRQ signalling on this interface instead of
out-of-band interrupt from brcmfmac

Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Signed-off-by: Erwan SZYMANSKI <erwan.szymanski@st.com>
Change-Id: I5986d3a245da37da26adb5bb747d99ac17a3dc29
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/385353
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm/boot/dts/stm32mp157c-dk2.dts | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/arch/arm/boot/dts/stm32mp157c-dk2.dts
+++ b/arch/arm/boot/dts/stm32mp157c-dk2.dts
@@ -129,6 +129,7 @@
 	pinctrl-1 = <&sdmmc2_b4_od_pins_a>;
 	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
 	non-removable;
+	cap-sdio-irq;
 	st,neg-edge;
 	bus-width = <4>;
 	vmmc-supply = <&v3v3>;
@@ -140,9 +141,6 @@
 	brcmf: bcrmf@1 {
 		reg = <1>;
 		compatible = "brcm,bcm4329-fmac";
-		interrupt-parent = <&gpiod>;
-		interrupts = <0 IRQ_TYPE_LEVEL_HIGH>; /* WL_HOST_WAKE */
-		interrupt-names = "host-wake";
 	};
 };
 
